solar/core/resource/virtual_resource.py

```diff
@@ -293,12 +296,32 @@ def parse_dict_input(r_input, args):
     return connections, assignments
 
 
+def add_assignment(assignments, r_input, arg):
+    try:
+        assignments[r_input].append(arg)
+    except KeyError:
+        assignments[r_input] = [arg]
+
+
 def is_connection(arg):
     if isinstance(arg, basestring) and '::' in arg:
         return True
     return False
 
 
+def parse_dicts_as_collection(child_input, arg):
```

Comment:
We will have now `parse_dict_input` and `parse_dicts_as_collection`. Can you merge them?